File: rust/fib.rs
// Iterative fibonacci up to 64-bit unsigned int
fn fib(n : u64) -> u64 {
  if n <= 1 { return n }
  else {
    // Assign these separately because of dead assignment warnings
    let mut tmp:u64;
    let (mut cur, mut next) = (0u64, 1u64);

    for _ in range(0, n) {
      tmp = cur + next;

      // Detect int overflow and halt
      if next < cur {
        println!("u64 overflow at n {:u}", n);
        return 0;
      }

      cur = next;
      next = tmp;
    }

    return cur;
  }
}

fn main() {
  // Just output fibonacci numbers 1-100
  // Spoiler alert: overflows at i=94
  for i in range(1u64,100u64) {
    println!("{} | {}", i, fib(i));
  }
}

File: go/fib.go
package main

import (
  "fmt"
)

// Iterative Fibonacci up to 'long'
func Fib(n int) int {
  if n <= 1 {
    return n
  } else {

    // some utility variables
    tmp, cur, next := 0, 0, 1

    for i := 0; i <= n; i++ {
      tmp = cur + next

      // Look for integer overflow
      // panic() if it's found
      if next < cur {
        fmt.Println("int overflow at n=", i)
        panic("integer overflow detected in Fib()")
      }

      cur = next
      next = tmp
    }

    return cur
  }
}

// Try to Fibonacci up to 100
// You can't, but try anyway
func main() {
  for i := 1; i <= 100; i++ {
    fmt.Println(i, " | ", Fib(i))
  }
}
